build.gradle.kts

```diff
@@ -44,29 +46,37 @@ sourceSets {
     }
 }
 
-val integrationTestImplementation by configurations.getting { extendsFrom(configurations.implementation.get()) }
-val integrationTestRuntimeOnly: Configuration by configurations.getting
+val integrationTestSourceSet: SourceSet = sourceSets["integrationTest"]
 
-configurations["integrationTestRuntimeOnly"].extendsFrom(configurations.runtimeOnly.get())
+val integrationTestImplementation: Configuration by
+    configurations.getting { extendsFrom(configurations.implementation.get()) }
+val integrationTestRuntimeOnly: Configuration by
+    configurations.getting { extendsFrom(configurations.runtimeOnly.get()) }
 
-val integrationTest =
+val integrationTestTask: Task =
     task<Test>("integrationTest") {
-        description = "Runs integration tests requiring external MongoDB deployment"
-        group = "verification"
+        group = LifecycleBasePlugin.VERIFICATION_GROUP
+        testClassesDirs = integrationTestSourceSet.output.classesDirs
+        classpath = integrationTestSourceSet.runtimeClasspath
+    }
 
-        testClassesDirs = sourceSets["integrationTest"].output.classesDirs
-        classpath = sourceSets["integrationTest"].runtimeClasspath
-        shouldRunAfter("test")
+tasks.check { dependsOn(integrationTestTask) }
 
-        useJUnitPlatform()
+tasks.withType<Test>().configureEach {
+    useJUnitPlatform()
+    testLogging { events(TestLogEvent.PASSED, TestLogEvent.SKIPPED, TestLogEvent.FAILED) }
+}
 
-        testLogging { events("passed") }
+// https://youtrack.jetbrains.com/issue/IDEA-234382/Gradle-integration-tests-are-not-marked-as-test-sources-resources
+idea {
+    module {
+        testSources.from(integrationTestSourceSet.allSource.srcDirs)
+        testResources.from(integrationTestSourceSet.resources.srcDirs)
```

Comment:
This is supposed to be needed only for Gradle < 7.3, as the bug was fixed in 7.3. But that, apparently, is not the case.

Reply:
that might be because of some misleading comments in the bug ticket, but the issue is still there for sure.